### PR TITLE
feat(mongodb): add support for v1alpha1

### PR DESCRIPTION
**Pull request by jremy42**

### CODE REVIEW

File: internal/services/mongodb/instance.go [hunk 1]
```diff
@@ -0,0 +1,405 @@
...
+			return diag.FromErr(err)
+		}
+	}
+
+	req := &mongodb.UpdateInstanceRequest{
+		Region:     region,
+		InstanceID: ID,
+	}
+
+	if d.HasChange("name") {
```
Comment (remyleone) [inline internal/services/mongodb/instance.go:332]: Introduce a variable that is changed like `should_update` or `has_changed` and if this variable is set then perform the update. That way the conditional line 342 won't get bigger

File: internal/services/mongodb/instance.go [hunk 2]
```diff
@@ -0,0 +1,405 @@
...
+		Region:     region,
+		InstanceID: ID,
+	}
+
+	if d.HasChange("password") {
+		password := d.Get("password").(string)
+		updateUserRequest.Password = &password
+	}
+
+	if updateUserRequest.Password != nil {
```
Comment (remyleone) [inline internal/services/mongodb/instance.go:364]: Same here, introduce a variable and check this variable to ensure than an update should be performed

File: internal/services/mongodb/snapshot.go [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+			Update:  schema.DefaultTimeout(defaultMongodbSnapshotTimeout),
+			Delete:  schema.DefaultTimeout(defaultMongodbSnapshotTimeout),
+			Default: schema.DefaultTimeout(defaultMongodbSnapshotTimeout),
+		},
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		SchemaVersion: 0,
+		Schema: map[string]*schema.Schema{
+			"id": {
```
Comment (remyleone) [inline internal/services/mongodb/snapshot.go:35]: Might be confusing with the SetId method. Why is this required?

File: internal/services/mongodb/snapshot.go [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	updateReq := &mongodb.UpdateSnapshotRequest{
+		SnapshotID: snapshotID,
+		Region:     region,
+	}
+
+	if d.HasChange("name") {
```
Comment (remyleone) [inline internal/services/mongodb/snapshot.go:167]: Introduce variable to trigger a change if this boolean value changed. Like `has_changed`

File: internal/services/mongodb/instance.go [hunk 5]
```diff
@@ -0,0 +1,405 @@
...
+			"node_type": {
+				Type:             schema.TypeString,
+				Required:         true,
+				Description:      "Type of node to use for the instance",
+				DiffSuppressFunc: dsf.IgnoreCase,
+			},
+			"user_name": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
```
[the remark this answers is not in the document]
Comment (jremy42) [reply internal/services/mongodb/instance.go:67]: Yes, the user_name cannot be updated at the moment, so changing it will indeed trigger the recreation of the whole instance.

[the remark this answers is not in the document]
Comment (jremy42) [reply internal/services/mongodb/instance.go:67]: doing nothing be a better solution?

File: internal/services/mongodb/snapshot.go (hunk 3 above)
Comment (jremy42) [reply internal/services/mongodb/snapshot.go:35]: You are right

File: docs/data-sources/mongodb_instance.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+- `region` - (Defaults to [provider](../index.md#region) `region`) The [region](../guides/regions_and_zones.md#zones) in which the MongoDB® Instance exists.
+
+- `organization_id` - (Defaults to [provider](../index.md#organization_id) `organization_id`) The ID of the organization the MongoDB® instance is in.
+
+## Attributes Reference
+
+In addition to all above arguments, the following attributes are exported:
+
+- `id` - The ID of the MongoDB® Instance.
```
Comment (remyleone) [inline docs/data-sources/mongodb_instance.md:50]: Will this ID still be present?

File: internal/services/mongodb/instance.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (remyleone) [reply internal/services/mongodb/instance.go:68]: Maybe having a dedicated resource for user, it is early but I would feel more comfortable with that rather than having a forcenew that could have large consequences if handled badly

File: docs/data-sources/mongodb_instance.md (hunk 6 above)
Comment (jremy42) [reply docs/data-sources/mongodb_instance.md:50]: That's normal, id is always exported, right?

**Review by remyleone — APPROVED** [review]
